cpp/benchmarks/ast/ast_benchmark.cu

```diff
@@ -0,0 +1,396 @@
+    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
+    cudf::ast::compute_column(table, expression_tree_root);
+  }
+
+  // Use the number of bytes read from global memory
+  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * state.range(0) *
+                          (tree_levels + 1) * sizeof(key_type));
+}
+
+#define AST_TRANSFORM_BENCHMARK_DEFINE(name, key_type, tree_type, reuse_columns) \
+  BENCHMARK_TEMPLATE_DEFINE_F(AST, name, key_type, tree_type, reuse_columns)     \
+  (::benchmark::State & st) { BM_ast_transform<key_type, tree_type, reuse_columns>(st); }
+
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int32_full_unique, int32_t, TreeType::FULL_COMPLETE, false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int64_full_unique, int64_t, TreeType::FULL_COMPLETE, false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_float_full_unique, float, TreeType::FULL_COMPLETE, false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_double_full_unique, double, TreeType::FULL_COMPLETE, false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int32_imbalanced_unique,
+                               int32_t,
+                               TreeType::IMBALANCED_LEFT,
+                               false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int64_imbalanced_unique,
+                               int64_t,
+                               TreeType::IMBALANCED_LEFT,
+                               false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_float_imbalanced_unique,
+                               float,
+                               TreeType::IMBALANCED_LEFT,
+                               false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_double_imbalanced_unique,
+                               double,
+                               TreeType::IMBALANCED_LEFT,
+                               false);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int32_full_reuse, int32_t, TreeType::FULL_COMPLETE, true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int64_full_reuse, int64_t, TreeType::FULL_COMPLETE, true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_float_full_reuse, float, TreeType::FULL_COMPLETE, true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_double_full_reuse, double, TreeType::FULL_COMPLETE, true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int32_imbalanced_reuse,
+                               int32_t,
+                               TreeType::IMBALANCED_LEFT,
+                               true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_int64_imbalanced_reuse,
+                               int64_t,
+                               TreeType::IMBALANCED_LEFT,
+                               true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_float_imbalanced_reuse, float, TreeType::IMBALANCED_LEFT, true);
+AST_TRANSFORM_BENCHMARK_DEFINE(ast_double_imbalanced_reuse,
+                               double,
+                               TreeType::IMBALANCED_LEFT,
+                               true);
+
+BENCHMARK_REGISTER_F(AST, ast_int32_imbalanced_unique)
+  ->Unit(benchmark::kMillisecond)
+  ->Args({100'000, 1})
```

Comment:
Use `->Ranges` to reduce all these Args lines to a single line.  Or you can define a custom function to generate ranges and call it with `->Apply(func)`.